mxcubeweb/core/adapter/beamline_action_adapter.py

```diff
@@ -4,19 +4,56 @@
 
 from mxcubecore.BaseHardwareObjects import HardwareObject
 from mxcubecore.HardwareObjects.BeamlineActions import BeamlineActions
+from pydantic.v1 import (
+    BaseModel,
+)
 
 from mxcubeweb.core.adapter.adapter_base import AdapterBase
+from mxcubeweb.core.components.queue import (
+    FAILED,
+    READY,
+    RUNNING,
+)
 from mxcubeweb.core.models.adaptermodels import (
+    BeamlineActionInputModel,
     HOActuatorValueChangeModel,
     NStateModel,
 )
 from mxcubeweb.core.models.configmodels import AdapterResourceHandlerConfigModel
 
 resource_handler_config = AdapterResourceHandlerConfigModel(
-    commands=["stop"], attributes=["data"]
+    commands=["stop", "run_action"], attributes=["data", "get_all_actions"]
 )
 
 
+class Argument(BaseModel):
```

Comment:
These models are only used by the BeamlineActions, I was hesitating of moving them to `models/adaptermodels` but decided to keep them in the same module for simplicity.